lang/cs/Org.Apache.REEF.Driver.Tests/TestTaskStates.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Driver.Task;
+using Xunit;
+
+namespace Org.Apache.REEF.Driver.Tests
+{
+    /// <summary>
+    /// The test cases in this classes test TaskState and transitions
+    /// </summary>
+    public class TestTaskStates
+    {
+        [Fact]
+        public void TestNewToCompleteSenario()
+        {
+            var taskState = new DriverTaskState();
+            Assert.True(taskState.CurrentState.Equals(TaskTransitionState.TaskNew));
+            Assert.True(taskState.MoveNext(TaskEvent.SubmittedTask).Equals(TaskTransitionState.TaskSubmitting));
+            Assert.True(taskState.MoveNext(TaskEvent.RunningTask).Equals(TaskTransitionState.TaskRunning));
+            Assert.True(taskState.MoveNext(TaskEvent.CompletedTask).Equals(TaskTransitionState.TaskCompleted));
+        }
+
+        [Fact]
+        public void TestRunningToCloseSenario()
+        {
+            var taskState = new DriverTaskState();
+            Assert.True(taskState.CurrentState.Equals(TaskTransitionState.TaskNew));
```

Comment:
A lot of Asserts seem excessive to me. For example, transition new-submitted-running is covered by 4 tests. Maybe we should verify this sequence in first test only, and in the next ones we should verify only steps which are unique to these tests?